### PR TITLE
add jacoco and jacoco-with-min-coverage-check modules working with maven/gradle

### LINKED ISSUE #8625: Module jacoco-check-min-coverage should work with gradle
The module is currently doing a mandatory replacement on `pom.xml` and don't use any maven plugin API of  `JHipsterModuleBuilder`:
https://github.com/jhipster/jhipster-lite/blob/5ba4d5674209f7f040b5ea9d0bc0f4d651f47290/src/main/java/tech/jhipster/lite/generator/server/javatool/jacoco/domain/JacocoThresholdModuleFactory.java#L16

We can convert the mandatory replacement in the `pom.xml` to an optional replacement, and add a new second optional replacement for `build.gradle.kts`. But this is not very robust, because the replacement can fail silently.

Another option is to use the maven / gradle plugin addition API of  `JHipsterModuleBuilder`, but the underlying java build commands should either be able to replace existing plugin presence to avoid duplicated plugin declaration, or the module should drop the plugin the re-add it with the new configuration.

### PR DESCRIPTION
**Pull request by renanfranca**
Fix #8625

### CODE REVIEW

File: src/main/java/tech/jhipster/lite/generator/server/javatool/jacoco/infrastructure/primary/JacocoModuleConfiguration.java [hunk 1]
```diff
@@ -10,16 +10,27 @@
...
+class JacocoModuleConfiguration {
 
   @Bean
-  JHipsterModuleResource jacocoThresholdModule(JacocoApplicationService jacoco) {
+  JHipsterModuleResource jacocoModule(JacocoApplicationService jacoco) {
     return JHipsterModuleResource.builder()
-      .slug(JACOCO_CHECK_MIN_COVERAGE)
+      .slug(JACOCO)
+      .withoutProperties()
+      .apiDoc("Java", "Add JaCoCo basic configuration")
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/generator/server/javatool/jacoco/infrastructure/primary/JacocoModuleConfiguration.java:20]:
```suggestion
      .apiDoc("Java", "Add JaCoCo for code coverage reporting")
```

File: src/main/java/tech/jhipster/lite/generator/server/javatool/jacoco/infrastructure/primary/JacocoModuleConfiguration.java [hunk 2]
```diff
@@ -10,16 +10,27 @@
...
+      .tags("server", "tools", "coverage")
+      .factory(jacoco::buildJacocoModule);
+  }
+
+  @Bean
+  JHipsterModuleResource jacocoWithMinCoverageCheckModule(JacocoApplicationService jacoco) {
+    return JHipsterModuleResource.builder()
+      .slug(JACOCO_WITH_MIN_COVERAGE_CHECK)
       .withoutProperties()
       .apiDoc("Java", "Add JaCoCo configuration to check minimum coverage")
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/generator/server/javatool/jacoco/infrastructure/primary/JacocoModuleConfiguration.java:31]:
```suggestion
      .apiDoc("Java", "Add JaCoCo for code coverage reporting and 100% coverage check")
```

File: src/main/java/tech/jhipster/lite/generator/server/javatool/jacoco/domain/JacocoModuleFactory.java [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+        .and()
+      .optionalReplacements()
+        .in(path("build.gradle.kts"))
+        .add(EXISTING_BUILD_GRADLE_TASK_TEST_NEEDLE, ADD_JACOCO_WITH_MIN_COVERAGE_CHECK_BUILD_GRADLE_TASK_TEST)
+        .and()
+      .and()
+      .optionalReplacements()
+        .in(path("pom.xml"))
+          .add(
+            lineAfterRegex(
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/generator/server/javatool/jacoco/domain/JacocoModuleFactory.java:107]: Not required anymore, isn't it?
It's replaced by the additional plugin execution added

Comment (renanfranca) [reply src/main/java/tech/jhipster/lite/generator/server/javatool/jacoco/domain/JacocoModuleFactory.java]: Thank you for the review, I pushed a new version 😄

File: tests-ci/generate.sh [hunk 4]
```diff
@@ -131,22 +131,14 @@ elif [[ $application == 'fullstack' ]]; then
 elif [[ $application == 'gradleapp' ]]; then
   init_server
   sonar_back
-  # TODO: use "spring_boot", "spring_boot_mvc" instead of the following individual modules
```
Comment (murdos) [inline tests-ci/generate.sh:134]: :tada:

File: src/main/java/tech/jhipster/lite/generator/slug/domain/JHLiteModuleSlug.java [hunk 5]
```diff
@@ -54,7 +54,8 @@ public enum JHLiteModuleSlug implements JHipsterModuleSlugFactory {
   INFINITEST_FILTERS("infinitest-filters"),
   INIT("init"),
   INTERNATIONALIZED_ERRORS("internationalized-errors"),
-  JACOCO_CHECK_MIN_COVERAGE("jacoco-check-min-coverage"),
+  JACOCO("jacoco"),
+  JACOCO_WITH_MIN_COVERAGE_CHECK("jacoco-with-min-coverage-check"),
```
Comment (murdos) [inline src/main/java/tech/jhipster/lite/generator/slug/domain/JHLiteModuleSlug.java:58]: Could you also update the script that generates our sample application? 
See https://github.com/jhipster/jhipster-lite-sample-app/blob/2537bfe1c34ef0bccf4fa229063789e6f43ff090/generate.sh#L65

**Review by murdos — APPROVED** [review]

File: tests-ci/generate.sh (hunk 4 above)
Comment (renanfranca) [reply tests-ci/generate.sh:134]: Great milestone!!!

File: src/main/java/tech/jhipster/lite/generator/slug/domain/JHLiteModuleSlug.java (hunk 5 above)
Comment (renanfranca) [reply src/main/java/tech/jhipster/lite/generator/slug/domain/JHLiteModuleSlug.java:58]: Done here: https://github.com/jhipster/jhipster-lite-sample-app/pull/3

**Comment by renanfranca** [comment]
@pascalgrimaud: Bounty Claimed https://opencollective.com/generator-jhipster/expenses/190423. Thanks 😄 ✌

**Comment by pascalgrimaud** [comment]
@renanfranca : approved

**Pull request merged**